great_expectations/datasource/fluent/interfaces.py

```diff
@@ -159,6 +162,7 @@ class DataAsset(FluentBaseModel, Generic[_DatasourceT]):
     id: Optional[uuid.UUID] = Field(default=None, description="DataAsset id")
 
     order_by: List[Sorter] = Field(default_factory=list)
+    batch_metadata: Optional[BatchMetadata] = pydantic.Field(default_factory=dict)
```

Comment:
We can remove the `Optional` here. If the user doesn't pass this in the `default_factory` will make an empty dict.

Reply:
For some reason I thought this would effect the schemas, but it doesn't so I made this change. ✅

Reply:
Have you pushed this change? 
To me it still shows up.
```suggestion
    batch_metadata: BatchMetadata = pydantic.Field(default_factory=dict)
```

Reply:
So the issue is that in-practice the `batch_metadata` gets passed in via `add` method as `Optional[BatchMetadata]`. Unless we handle the `None` case, the type checker will not allow this field to be strict `BatchMetadata` with a `dict` default. Handling the `None` case would require us to concretely define the `add` methods on all datasources and perform the type conversion. You can't do the type conversion in a validator and have `mypy` pick up on the types correctly (see: https://github.com/pydantic/pydantic/discussions/5050).

Reply:
@billdirks What this also means is that `order_by` is not currently working for any Datasources other than `SQLDatasource`. For `order_by` we do type conversion in the `SQLDatasource` `add` methods, but nowhere else.

Reply:
But `batch_metadata` gets passed in via kwargs in the implementation, doesn't it? So it won't actually be non it just won't be provided???

Exception being for `pandas_default` because those are concrete `add_<ASSET>` method.

Reply:
I am changing `batch_metadata` to always be passed in via `kwargs` in a follow-up PR. The comments about `order_by` may not actually be founded, because it does look like we are doing type conversion in the `order_by` validator, so if it gets passed in via `kwargs` it should work.

Reply:
The `order_by` stubs are "covering up" what would typically get surfaced as a typing error because the validator typing information does not get passed along.